runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb

```diff
@@ -41,10 +41,12 @@ def get_long_running_operation_result(azure_response, custom_deserialization_blo
       if !AsyncOperationStatus.is_terminal_status(polling_state.status)
         task = Concurrent::TimerTask.new do
           begin
-            if !polling_state.azure_async_operation_header_link.nil?
-              update_state_from_azure_async_operation_header(polling_state.get_request(headers: request.headers, base_uri: request.base_uri, user_agent_extended: user_agent_extended), polling_state)
-            elsif !polling_state.location_header_link.nil?
-              update_state_from_location_header(polling_state.get_request(headers: request.headers, base_uri: request.base_uri, user_agent_extended: user_agent_extended), polling_state, custom_deserialization_block)
+            if !polling_state.azure_async_operation_header_link.nil? || !polling_state.location_header_link.nil?
```

Comment:
Does this condition means that now as long as you find either `azure_async_operation_header_link` or `location_header_link` with `custom_deserialization_block` you'll call `update_state_with_custom_deserialization`? Because if i understand correctly, we were only calling update_state_without_custom_deserialization (old name: update_state_from_azure_async_operation_header) only when we have `azure_async_operation_header_link` not when we find `location_header_link`?